build.sh

```diff
@@ -0,0 +1,548 @@
+  CMAKE_BASIC_ARGS="-DCOORD_TYPE=$COORD"
+
+  # Add configuration-specific arguments
+  if [[ -n "$STATIC" && "$STATIC" == "1" ]]; then
+    CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DBUILD_STATIC=ON"
+  fi
+
+  if [[ -n "$LITE" && "$LITE" == "1" ]]; then
+    CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DBUILD_LITE=ON"
+  fi
+
+  if [[ "$BUILD_TESTS" == "1" ]]; then
+    CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DBUILD_SEARCH_UNIT_TESTS=ON"
+  fi
+
+  if [[ -n "$SAN" ]]; then
+    CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DSAN=$SAN"
+    DEBUG="1"
+  fi
+
+  # Set build type
+  if [[ "$DEBUG" == "1" ]]; then
+    CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DCMAKE_BUILD_TYPE=Debug"
+  else
+    CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DCMAKE_BUILD_TYPE=Release"
+  fi
+
+  # Ensure output file is always .so even on macOS
+  CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -DCMAKE_SHARED_LIBRARY_SUFFIX=.so"
+
+  # Add caching flags to prevent using old configurations
+  CMAKE_BASIC_ARGS="$CMAKE_BASIC_ARGS -UCMAKE_TOOLCHAIN_FILE"
+}
+
+#-----------------------------------------------------------------------------
+# Function: run_cmake
+# Run CMake to configure the build
+#-----------------------------------------------------------------------------
+run_cmake() {
+  # Create build directory and ensure any parent directories exist
+  mkdir -p "$BINDIR"
+  cd "$BINDIR"
+
+  # Clean up any cached CMake configuration if force is enabled
+  if [[ "$FORCE" == "1" ]]; then
+    echo "Cleaning CMake cache..."
+    rm -f CMakeCache.txt
+    rm -rf CMakeFiles
+  fi
+
+  echo "Configuring CMake..."
+  echo "Build directory: $BINDIR"
+
+  # Run CMake with all the flags
+  if [[ "$FORCE" == "1" || ! -f "$BINDIR/Makefile" ]]; then
+    CMAKE_CMD="cmake $ROOT $CMAKE_BASIC_ARGS $CMAKE_ARGS"
+    echo "$CMAKE_CMD"
+    
+    # If verbose, dump all CMake variables before and after configuration
+    if [[ "$VERBOSE" == "1" ]]; then
+      echo "Running CMake with verbose output..."
+      $CMAKE_CMD --trace-expand
+    else
+      $CMAKE_CMD
+    fi
+  fi
+}
+
+#-----------------------------------------------------------------------------
+# Function: build_redisearch_rs
+# Build the redisearch_rs target explicitly
+#-----------------------------------------------------------------------------
+build_redisearch_rs() {
+  echo "Building redisearch_rs..."
+  REDISEARCH_RS_DIR="$ROOT/src/redisearch_rs"
+  REDISEARCH_RS_TARGET_DIR="$ROOT/bin/redisearch_rs"
+  REDISEARCH_RS_BINDIR="$BINDIR/redisearch_rs"
+
+  # Determine Rust build mode
+  if [[ "$DEBUG" == "1" ]]; then
+    RUST_BUILD_MODE=""
+    RUST_ARTIFACT_SUBDIR="debug"
+  else
+    RUST_BUILD_MODE="--release"
+    RUST_ARTIFACT_SUBDIR="release"
+  fi
+
+  # Build using cargo
+  mkdir -p "$REDISEARCH_RS_TARGET_DIR"
+  cd "$REDISEARCH_RS_DIR"
+  cargo build $RUST_BUILD_MODE
+
+  # Copy artifacts to the target directory
+  mkdir -p "$REDISEARCH_RS_BINDIR"
+  cp "$REDISEARCH_RS_TARGET_DIR/$RUST_ARTIFACT_SUBDIR"/*.a "$REDISEARCH_RS_BINDIR"
+}
```

Comment:
@LukeMathWalker - can you look here and see if this is equivalent to what we have today in the makefile?

Reply:
Unit tests and coverage tests are still running via `makefile` (will come in next PR). Only the basic build and for sanitizer should be covered here. In which case, is that OK, or changes still required?

Reply:
what is the conclusion here? can we proceed?

Reply:
@LukeMathWalker ?